pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -579,6 +594,7 @@ func (cb *ClusterBuilder) buildBlackHoleCluster() *cluster.Cluster {
 		ConnectTimeout:       gogo.DurationToProtoDuration(cb.push.Mesh.ConnectTimeout),
 		LbPolicy:             cluster.Cluster_ROUND_ROBIN,
 	}
+	cb.applyMetadataExchange(c)
```

Comment:
Do we need MX for blackhole cluster? I know we probably add it via EnvoyFilter before, but does not seem to be necessary?

Reply:
Good point. There were a lot of questionable filters in EnvoyFilter logic, I just matched them currently since it made testing a lot easier (diff of configdump). Open to chaning it though; this seems like a case where it clearly doesn't matter

Reply:
With EnvoyFilter, we lack of fine granular control on what to insert and what not. Since we are implementing it first class, we should do it right instead of matching EnvoyFilter.